galsim/celestial.py

```diff
@@ -582,7 +585,79 @@ def galactic(self, epoch=2000.):
+        the transformation, the obliquity of the ecliptic (the Earth's axial tilt).
+
+        @param  epoch        The epoch to be used for estimating the obliquity of the ecliptic.
+                             [default: 2000.]
+        @param  julian_date  The Julian date to be used to calculate the ecliptic coordinates.  If
+                             None, then the ecliptic coordinates that are returned should be
+                             interpreted as those with respect to the sun (i.e., assuming the sun is
+                             at (0,0) as it is at the vernal equinox).
+                             [default: None]
+
+        @returns the longitude and latitude as a tuple (lambda, beta), given as Angle instances.
+        """
+        import math
+
+        # We are going to work in terms of the (x, y, z) projections.
+        self._set_aux()
+
+        # We need to figure out the time in Julian centuries from J2000 for this epoch.
+        t = (epoch - 2000.)/100.
+        # Then we use the last (most recent) formula listed under
+        # http://en.wikipedia.org/wiki/Ecliptic#Obliquity_of_the_ecliptic, from
+        # JPL's 2010 calculations.
+        ep = galsim.DMS_Angle('23:26:21.406')
+        ep -= galsim.DMS_Angle('00:00:46.836769')*t
+        ep -= galsim.DMS_Angle('00:00:0.0001831')*(t**2)
+        ep += galsim.DMS_Angle('00:00:0.0020034')*(t**3)
+        # There are even higher order terms, but they are really not important for any reasonable
+        # calculation we could ever do with GalSim.
+
+        cos_ep = math.cos(ep.rad())
+        sin_ep = math.sin(ep.rad())
+
+        # Coordinate transformation here, from celestial to ecliptic:
+        x_ecl = self._x
+        y_ecl = cos_ep*self._y + sin_ep*self._z
+        z_ecl = -sin_ep*self._y + cos_ep*self._z
+
+        beta = math.asin(z_ecl)*galsim.radians
+        lam = math.atan2(y_ecl, x_ecl)*galsim.radians
+
+        if julian_date is not None:
+            lam_sun, _ = _sun_position_ecliptic(julian_date)
+            lam += lam_sun
+
+        return (lam, beta)
+
     def copy(self): return CelestialCoord(self._ra, self._dec)
 
     def __repr__(self): return 'CelestialCoord('+repr(self._ra)+','+repr(self._dec)+')'
 
+def _sun_position_ecliptic(date):
+    # This is a helper routine to calculate the position of the sun in ecliptic coordinates given a
+    # Julian date.  It is most precise for dates between 1950-2050, and is based on
+    # http://en.wikipedia.org/wiki/Position_of_the_Sun#Ecliptic_coordinates
+    # We start by getting the number of days since Greenwich noon on 1 January 2000 (J2000).
+    import math
+    n = date - 2451545.0
+    L = (280.46*galsim.degrees + (0.9856474*galsim.degrees)*n).wrap()
+    g = (357.528*galsim.degrees + (0.9856003*galsim.degrees)*n).wrap()
+    lam = L + (1.915*galsim.degrees)*math.sin(g.rad()) + \
+        (0.020*galsim.degrees)*math.sin((2*g).rad())
+    lam += 360.*galsim.degrees
```

Comment:
I believe this line doesn't do anything given the `wrap()` call in the next line.